services/navigation/builtin/builtin.go

```diff
@@ -581,5 +624,81 @@ func (svc *builtIn) Obstacles(ctx context.Context, extra map[string]interface{})
 func (svc *builtIn) Paths(ctx context.Context, extra map[string]interface{}) ([]*navigation.Path, error) {
 	svc.mu.RLock()
 	defer svc.mu.RUnlock()
-	return nil, errUnimplemented
+
+	rawExecutionWaypoint := svc.activeExecutionWaypoint.Load()
+	// If there is no execution, return empty paths
+	if rawExecutionWaypoint == nil || rawExecutionWaypoint == emptyExecutionWaypoint {
+		return []*navigation.Path{}, nil
+	}
+
+	ewp, ok := rawExecutionWaypoint.(executionWaypoint)
+	if !ok {
+		return nil, errors.New("execution corrupt")
+	}
+
+	ph, err := svc.motionService.PlanHistory(ctx, motion.PlanHistoryReq{
+		ComponentName: svc.base.Name(),
+		ExecutionID:   ewp.executionID,
+		LastPlanOnly:  true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	geoPoints := make([]*geo.Point, 0, len(ph[0].Plan.Steps))
+	for _, s := range ph[0].Plan.Steps {
+		if len(s) > 1 {
+			return nil, errors.New("multi component paths are unsupported")
+		}
+		var pose spatialmath.Pose
+		for _, p := range s {
+			pose = p
+		}
+
+		geoPoint := geo.NewPoint(pose.Point().Y, pose.Point().X)
+		geoPoints = append(geoPoints, geoPoint)
+	}
+
+	path, err := navigation.NewPath(ewp.waypoint.ID.String(), geoPoints)
+	if err != nil {
+		return nil, err
+	}
+	return []*navigation.Path{path}, nil
+}
+
+func pollUntilMOGSuccessOrError(
+	ctx context.Context,
+	m motion.Service,
+	interval time.Duration,
+	req motion.PlanHistoryReq,
+) error {
+	for {
+		ph, err := m.PlanHistory(ctx, req)
+		if err != nil {
+			return err
+		}
+
+		status := ph[0].StatusHistory[0]
+
+		switch status.State {
+		case motion.PlanStateInProgress:
+		case motion.PlanStateFailed:
+			err := errors.New("plan failed")
+			if reason := status.Reason; reason != nil {
+				err = errors.New(*reason)
```

Comment:
Maybe we want to wrap the reason with the "plan failed"?  I'm not sure what form the reasons take so it might be that we already have this

Reply:
done

Reply:
also it looks like this: `error state exceeds planDeviationMM; planDeviationMM: 1000.000000, errorstate.Point().Norm(): 1056.301307, errorstate.Point(): r3.Vector{X:-1037.7992823209859, Y:-196.83775392595362, Z:0} ` before the wrap